bip-0360.mediawiki

```diff
@@ -1,705 +1,401 @@
+
+Some may balk at the threat of quantum computers to Bitcoin given their limited functionality to date. Others – including governments, corporations, and existing and potential Bitcoin users – are concerned about the potential for their advancement and future use in cryptanalysis.
+
+The Commercial National Security Algorithm Suite (CNSA) 2.0, for instance, has mandated software and networking equipment to be upgraded to post-quantum schemes by 2030, with browsers and operating systems fully upgraded by 2033. Additionally, according to NIST IR 8547, Elliptic Curve Cryptography is planned to be disallowed within the US federal government after 2035 (with an exception made for hybrid cryptography, or the use of ECC and post-quantum algorithms together). These kinds of mandates have triggered concern by some ECC users – including risk-averse Bitcoin users who prefer to be prepared out of an abundance of caution. While it is unclear when or if CRQCs will become viable, we propose the addition of a Tapscript-native, hashed output type for those interested in enhanced resistance to this threat.
+
+Even in the most optimistic case – wherein quantum computers never pose a significant risk to ECC – we believe that public fear of their advancement alone may be influencing adoption and broad confidence in the Bitcoin network. In other words, we believe users’ fear of quantum computers may be worth addressing regardless of CRQC viability, which is difficult to assess. Given these concerns, we think it’s worth considering changes that offer quantum-resistance, are minimal in complexity and risk – and that create new options for using Bitcoin (and Tapscript).
+
+As a conservative first step in this effort, we propose Pay-to-Tapscript-Hash (P2TSH) – a Tapscript-native, hashed output type that can be used in a quantum-resistant manner.
+
+===Long-Exposure vs Short-Exposure Quantum Vulnerability===
+
+This proposal specifically mitigates against the risk of long-exposure attacks on Taproot outputs. While other (hashed) Bitcoin output script types already mitigate against this risk, Taproot outputs expose their public keys and so are currently vulnerable to long-exposure quantum attacks.
+
+A long-exposure attack is an attack performed on exposed blockchain data, such as exposed public keys or the spend scripts of spent outputs. These are likely to be the earliest quantum attacks made possible on Bitcoin, because quantum attackers have ample time – as much time as vulnerable keys are exposed – to carry out quantum key recovery.
+
+Short-exposure attacks would likely require faster quantum computers with greater choerence and error-tolerance – because these attacks must occur within the relatively short time that a transaction is being confirmed in block, while the public key(s) are exposed in the mempool.
```

Comment:
```suggestion
Short-exposure attacks would likely require faster quantum computers  – because these attacks must occur within the relatively short time that a transaction is waiting to being confirmed in the blockchain, while the public key(s) are exposed in the mempool.
```

I don't see why "with greater choerence [coherence] and error-tolerance" are needed  here. Faster/more powerful should be enough all things being equal. Greater error-tolerance would result in a more powerful quantum computer.